variables.tf

```diff
@@ -2,8 +2,150 @@
+  type        = string
+
+  validation {
+    condition     = var.prefix != "" && length(var.prefix) <= 25
+    error_message = "You must provide a value for the resource_prefix variable and the resource_prefix length can't exceed 25 characters."
+  }
+}
+
+variable "region" {
+  default     = "us-south"
+  description = "Region where the watsonx resources will be provisioned."
+  type        = string
+
+  validation {
+    condition     = contains(["eu-de", "us-south", "eu-gb", "jp-tok", "au-syd"], var.region)
+    error_message = "You must specify `eu-de`, `eu-gb`, `jp-tok` or `us-south` as the IBM Cloud region."
+  }
+}
+
+variable "resource_tags" {
+  description = "Optional list of tags to describe the service instances created by the module."
+  type        = list(string)
+  default     = []
+}
+
+# Watson Studio
+variable "existing_watson_studio_instance_crn" {
+  default     = null
+  description = "The CRN of an existing Watson Studio instance. If not provided, a new instance will be provisioned."
+  type        = string
+}
+
+variable "watson_studio_plan" {
+  default     = "professional-v1"
+  description = "The plan that is used to provision the Watson Studio instance. The plan you choose for Watson Studio affects the features and capabilities that you can use."
+  type        = string
+  validation {
+    condition     = contains(["free-v1", "professional-v1"], var.watson_studio_plan)
+    error_message = "You must use a free-v1 or professional-v1 plan. Learn more."
+  }
+}
+
+# Watson Machine Learning
+variable "existing_machine_learning_instance_crn" {
+  default     = null
+  description = "The CRN of an existing Watson Machine Learning instance. If not provided, a new instance will be provisioned."
+  type        = string
+}
+
+variable "watson_machine_learning_plan" {
+  description = "The plan that is used to provision the Watson Machine Learning instance."
+  type        = string
+  default     = "v2-professional"
+
+  validation {
+    condition     = contains(["lite", "v2-professional", "v2-standard"], var.watson_machine_learning_plan)
+    error_message = "The plan must be lite, v2-professional, or v2-standard. Learn more."
+  }
+}
+
+variable "watson_machine_learning_service_endpoints" {
+  type        = string
+  description = "The type of service endpoints. Possible values: 'public', 'private', 'public-and-private'."
+  default     = "public"
+
+  validation {
+    condition     = contains(["public", "public-and-private", "private"], var.watson_machine_learning_service_endpoints)
+    error_message = "The specified service endpoint is not valid. Supported options are public, public-and-private, or private."
+  }
+
+  validation {
+    condition     = contains(["lite"], var.watson_machine_learning_plan) ? contains(["public"], var.watson_machine_learning_service_endpoints) : true
+    error_message = "The lite plan of watson machine learning only supports public endpoints."
```

Comment:
Is this mentioned anywhere? If so, please provide a reference link to it else this can be removed.

Reply:
If we choose lite plan, then there is no option of endpoints to be selected and with other plans we can choose. So it should be there. Not finding doc link as of now, but thats fine.